src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -2904,18 +2915,58 @@ def forward(self, outputs, targets):
 
         if "enc_outputs" in outputs:
             enc_outputs = outputs["enc_outputs"]
-            bin_targets = copy.deepcopy(targets)
-            for bt in bin_targets:
-                bt["class_labels"] = torch.zeros_like(bt["class_labels"])
-            indices = self.matcher(enc_outputs, bin_targets)
+            indices = self.matcher(enc_outputs, targets)
             for loss in self.losses:
-                l_dict = self.get_loss(loss, enc_outputs, bin_targets, indices, num_boxes)
+                l_dict = self.get_loss(loss, enc_outputs, targets, indices, num_boxes)
                 l_dict = {k + "_enc": v for k, v in l_dict.items()}
                 losses.update(l_dict)
 
         return losses
 
 
+def build_label_maps(logits, input_ids):
+    """
+    Computes a mapping between the tokens associated with the prompt labels in the logit space with shape `(batch_size, num_labels, hidden_size)`
+    where `num_labels` is defined by the number of classes in the input prompt.
+
+    For instance, given the prompt "fish. shark." we get input_ids = [  101,  3869,  1012, 11420,  1012,   102].
+    This function will return a mapping for each of the prompt tokens (i.e. tokens associated with "fish" and "shark")
+    indicating their position in the logit space.
```

Comment:
Based on this - I think you mean position rather than "space" i.e. this function extracts the token positions for input labels. Mapping to logit space implies a linear transformation which takes you from "prompt vector space" to "logic vector space"